sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/TPCDSQueryBenchmark.scala

```diff
@@ -70,11 +67,12 @@ object TPCDSQueryBenchmark extends SqlBasedBenchmark with Logging {
 
   def setupTables(dataLocation: String, createTempView: Boolean): Map[String, Long] = {
     tables.map { tableName =>
-      val df = spark.read.parquet(s"$dataLocation/$tableName")
       if (createTempView) {
-        df.createOrReplaceTempView(tableName)
+        spark.read.parquet(s"$dataLocation/$tableName").createOrReplaceTempView(tableName)
       } else {
-        df.write.saveAsTable(tableName)
+        spark.sql(s"DROP TABLE IF EXISTS $tableName")
+        spark.catalog.createTable(tableName, s"$dataLocation/$tableName", "parquet")
+        Try(spark.sql(s"ALTER TABLE $tableName RECOVER PARTITIONS"))
```

Comment:
Since this is a non-trivial, could you add a comment about the reason why we need `Try` here?

Reply:
Sure, added a comment in https://github.com/apache/spark/pull/31218/commits/976b19bf5a2c09dcc902d85d197cdde81e45f477.